### ISSUE TITLE
Modify pod command from manifest

### ISSUE THREAD

**Issue by juldrixx** [konpyutaika/nifikop#46]
From nifikop created by [jrmanes](https://github.com/jrmanes): Orange-OpenSource/nifikop#123  

## Feature Request

**Is your feature request related to a problem? Please describe.**
It will be great in order to change the command from the manifest due to I need to execute some commands inside the container.

**Describe the solution you'd like to see**
Add an option to check if there is command from the manifest (spec).

**Describe alternatives you've considered**
Doing it with an initcontainer, but it will have to mount the nifi home path.

**Comment by juldrixx** [comment]
perhaps `NiFiCluster.spec.extraArgs` could be added to run additional commands to be added here:

https://github.com/Orange-OpenSource/nifikop/blob/master/pkg/resources/nifi/pod.go#L452

I think it's a straightforward change.

**Comment by floppyfish14** [comment]
Was this ever implemented? 

I'm desperately trying to make nifi work on my cluster with very stringent security requirements. I cannot use any operator for nifi except this one and I cannot figure out a way to make this nifi operator start up my pods using the typical entrypoint of '../scripts/start.sh' so that the pod does not exit immediately after startup.

The built in entrypoint from here: https://github.com/konpyutaika/nifikop/blob/master/pkg/resources/nifi/pod.go#L534 Will not work for my setup.

**Comment by juldrixx** [comment]
> scripts/start.sh

Why does it exit after startup?

**Comment by mh013370** [comment]
A pod exiting immediately after startup would indicate some problem with the deployment configuration/environment. I would recommend taking a look at both nifikop logs and nifi-app.logs from one of the failing pods to see what is happening.

Do you know why the existing entrypoint won't worth for your setup? What needs to change?

**Comment by floppyfish14** [comment]
The issue I found was in the startup command that is hardcoded into the operator. I had to add a kyverno policy to my cluster to override the startup command for the spawned pods. The app would start up but then immediately and suddenly exit because the `exec bin/nifi.sh run` command would (for whatever reason) execute in the background. My cluster then assumed the pod died and would try to restart it. Another issue I had, was that I set the initial node to an ID of 0 and that caused the operator to kill it on repeat forever no matter the condition of the pod.

I also needed to overwrite the default readiness and liveness probes for my pods so the cluster would stop thinking they had failed to start.

Again, the cluster I am operating has to meet very high security standards and there are many things that most people will not run into in our config. I assume my case is an edge case. For others, here is my kyverno policy that I built into my helm chart.

{{- if .Values.nifiCluster.overrideEntrypoint }}
apiVersion: kyverno.io/v1
kind: ClusterPolicy
metadata:
  name: replace-nifi-pod-command
spec:
  rules:
    - name: replace-command-for-nifi-pods
      match:
        resources:
          kinds:
            - Pod
          namespaces:
            - nifi
          names:
            - "nifi-*-node*"
      mutate:
        patchStrategicMerge:
          spec:
            containers:
              - name: "nifi"
                imagePullPolicy: Always
                command:
                - bash
                - -ce
                - "cp ${NIFI_HOME}/tmp/* ${NIFI_HOME}/conf/ && bash ../scripts/start.sh"
{{- end }}
